frontend/packages/knative-plugin/src/components/knatify/CreateKnatifyPage.tsx

```diff
@@ -0,0 +1,137 @@
+  WatchK8sResults,
+  WatchK8sResultsObject,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import NamespacedPage, {
+  NamespacedPageVariants,
+} from '@console/dev-console/src/components/NamespacedPage';
+import DeployImageForm from '@console/dev-console/src/components/import/DeployImageForm';
+import { handleRedirect } from '@console/dev-console/src/components/import/import-submit-utils';
+import { DeployImageFormData } from '@console/dev-console/src/components/import/import-types';
+import { deployValidationSchema } from '@console/dev-console/src/components/import/deployImage-validation-utils';
+import {
+  getInitialValuesKnatify,
+  knatifyResources,
+  getKnatifyWorkloadData,
+} from '../../utils/knatify-utils';
+
+type watchResource = {
+  [key: string]: K8sResourceKind[] | K8sResourceKind;
+};
+
+type CreateKnatifyPageProps = RouteComponentProps<{ ns?: string }>;
+
+const CreateKnatifyPage: React.FunctionComponent<CreateKnatifyPageProps> = ({
+  match,
+  location,
+}) => {
+  const { t } = useTranslation();
+  const namespace = match.params.ns;
+  const queryParams = new URLSearchParams(location.search);
+  const kind = queryParams.get('kind');
+  const appName = queryParams.get('name');
+  const [perspective] = useActivePerspective();
+  const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+  const [loaded, setloaded] = React.useState<boolean>(false);
+
+  const watchedResources = React.useMemo(
+    () => ({
+      projects: {
+        kind: ProjectModel.kind,
+        isList: true,
+      },
+      imageStream: {
+        kind: 'ImageStream',
+        isList: true,
+        namespace,
+        selector: {
+          matchLabels: { 'app.kubernetes.io/instance': appName },
+        },
+        optional: true,
+      },
+      ...(kind &&
+        appName && {
+          workloadResource: {
+            kind,
+            name: appName,
+            namespace,
+            optional: true,
+          },
+        }),
+    }),
+    [namespace, kind, appName],
+  );
+
+  const resources: WatchK8sResults<watchResource> = useK8sWatchResources<watchResource>(
+    watchedResources,
+  );
+
+  React.useEffect(() => {
+    const resKeys = Object.keys(resources);
+    if (
+      resKeys.length > 0 &&
+      resKeys.every((key) => resources[key].loaded || !!resources[key].loadError)
+    ) {
+      setloaded(true);
+    }
+  }, [resources]);
+
+  const handleSubmit = (
+    values: DeployImageFormData,
+    helpers: FormikHelpers<DeployImageFormData>,
+  ) => {
+    return knatifyResources(values)
+      .then(() => {
+        helpers.setStatus({ submitError: '' });
+        handleRedirect(namespace, perspective, perspectiveExtensions);
+      })
+      .catch((err) => {
+        helpers.setStatus({ submitError: err.message });
+      });
+  };
+
+  return (
+    <NamespacedPage disabled variant={NamespacedPageVariants.light}>
+      <Helmet>
+        <title>{t('knative-plugin~Knatify')}</title>
```

Comment:
nit: better to have `Create Knative service` in title too.
```suggestion
        <title>{t('knative-plugin~Create Knative service')}</title>
```